core/templates/dev/head/services/StateRulesStatsService.js

```diff
@@ -59,15 +59,22 @@ oppia.factory('StateRulesStatsService', [
             visualizations_info: response.data.visualizations_info.map(
               function(vizInfo) {
                 var newVizInfo = angular.copy(vizInfo);
-                if (newVizInfo.addressed_info_is_supported) {
-                  newVizInfo.data.forEach(function(vizInfoDatum) {
+                // If data is a FractionInput, need to change data so that
+                // visualization displays the input in a readable manner.
+                newVizInfo.data.forEach(function(vizInfoDatum) {
+                  if (state.interaction.id === 'FractionInput') {
+                    vizInfoDatum.answer =
+                        FractionObjectFactory.fromDict(
+                          vizInfoDatum.answer).toString();
+                  }
+                  if (newVizInfo.addressed_info_is_supported) {
```

Comment:
That was there from before I changed the file, although I rearranged it a bit, as previously it had the if statement outside of the foreach loop. I just added the check inside the loop since I needed to loop through each of them anyway.

Reply:
Ah I see now, thanks!